### PR TITLE
tmp dir/file hooks using resources list

### PR DESCRIPTION
**Pull request by natechols**
A more minimal implementation.

### CODE REVIEW

File: pbcommand/models/tool_contract.py [hunk 1]
```diff
@@ -251,6 +251,22 @@ def __init__(self, task_id, is_distributed, input_files, output_files,
...
+    @property
+    def tmp_dir(self):
+        for resource in self.resources:
+            if "type_id" in resource:
+                if resource["type_id"] == ResourceTypes.TMP_DIR:
+                    return resource["path"]
+        return None
+
+    @property
+    def tmp_file(self):
```
Comment (mpkocher) [inline pbcommand/models/tool_contract.py:263]: In principle, these are lists. 

Also, if it's not clear, the resource 'types' in the RTC and the TC are different.

TC:

resources = ['$tmpdir', '$tmpfile', '$tmpfile']

In the resolved TC:

``` javascript
resources = [
{'type_id':'$tmpdir', 'path': "/path/to-dir"}, 
{"type_id": "$tmpfile", "path": "/path/to/t2.txt"}, 
("type_id": "$tmpfile", "path": "/path/to/t2.txt"}]
```

**Comment by natechols** [comment]
I still don't understand what you're asking me to do.  Is this code incorrect?

**Comment by natechols** [comment]
FWIW, without any additional changes to pbsmrtpipe it is now producing resolved tool contracts with this:

```
    "resources": [
        {
            "path": "/tmp",
            "type_id": "$tmpdir"
        }
    ],
```

so adding the rest of the functionality would just be a matter of making the tmp_dir variable user-controllable instead of having "/tmp" hardcoded.

**Comment by mpkocher** [comment]
It shouldn't be hardcoded. pbsmrtpipe.options.tmp_dir should set the tmp dir. It defaults to /tmp.

**Pull request merged**